### PR TITLE
New party

### LINKED ISSUE #6: New Viewing Party Page
As an authenticated user,
When I visit the new viewing party page,
I should see a form with the following:
- [x] Movie Title (that is un-editable)
- [x] Duration of Party with a default value of movie runtime in minutes
- [x] When: field to select date
- [x] Start Time: field to select time
- [x] Checkboxes next to each friend (if user has friends)
- [x]  Button to create a party

**Details** When the party is created, the authenticated user should be redirected back to the dashboard where the new event is shown. The event should also be seen by any friends that were invited when they log in.

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
I dont understand why my commits from a different branch are showing up on this pr it makes no sense, I need to ask a mentor about it.

This nearly completes the new viewing party story, however theres a few things I want to change around and a few tests to write once I do.

-Adds new party form
-Adds new and create actions in the parties controller, the create action creates partyusers rows for each friend that has been invited
-Adds relationships between users, parties, userparties
-Edits dashboard view to use movie object

Im unsure of passing the runtime and title in the params it seems kind of odd, but maybe ok?
I want to rework the show action for the dashboard as I feel all the ar queries might be unnecessary given all the relationships between users and parties.
Also Im really blown away by rails' ability to understand pluralization, wild.

### CODE REVIEW

**Comment by PhilipDeFraties** [comment]
I thought I had a good handle on how to separate different commits between branches but apparently I do not. Gotta get with my mentor to go over how to properly do this.

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -2,28 +2,17 @@ class MoviesController < ApplicationController
...
 
   def show
-    conn = Faraday.new(url: 'https://api.themoviedb.org') do |faraday|
-      faraday.headers['X-API-Key'] = ENV['MOVIEDB_API_KEY']
-    end
-
-    movie = conn.get("/3/movie/#{params[:id]}?api_key=#{ENV['MOVIEDB_API_KEY']}&language=en-US")
-    @movie_details = JSON.parse(movie.body, symbolize_names: true)
-
+    @movie = SearchFacade.find_movie(params[:id])
```
Comment (LHJE) [inline app/controllers/movies_controller.rb:5]: SUCH a good idea!

File: app/controllers/parties_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+                                          date: params[:party_date],
+                                          time: params[:start_time]
+                                        })
+
+    params[:invitees].each do |user_id|
+      party.party_users.create(party_id: party.id, user_id: user_id)
+    end
+    redirect_to '/user/dashboard'
+  end
+end
```
Comment (LHJE) [inline app/controllers/parties_controller.rb:21]: Love all this

File: app/controllers/user/dashboard_controller.rb [hunk 3]
```diff
@@ -1,23 +1,3 @@
...
-        'accepted'
-      elsif status == 1
-        'declined'
-      else
-        'pending'
-      end
-    end
-  end
+  def show; end
 end
```
Comment (LHJE) [inline app/controllers/user/dashboard_controller.rb:5]: Amazing. You got all that down to 3 lines haha

File: app/models/party.rb [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
             :time, presence: true
+
+  def user_status(user_id)
+    if self.user_id == user_id
+      'Host'
+    else
+      'Invited'
+    end
+  end
 end
```
Comment (LHJE) [inline app/models/party.rb:20]: Love this

File: app/models/party_user.rb [hunk 5]
```diff
@@ -1,4 +1,4 @@
 class PartyUser < ApplicationRecord
-  # has_many :users
-  # has_many :viewing_parties
+  belongs_to :user
+  belongs_to :party
 end
```
Comment (LHJE) [inline app/models/party_user.rb:4]: I knew we'd have to do something like this hah

File: app/models/user.rb [hunk 6]
```diff
@@ -7,5 +7,12 @@ class User < ApplicationRecord
...
   validates :password_confirmation, presence: { require: true }
 
+  has_many :friendships
   has_many :friends, through: :friendships, source: :friend
+
+  has_many :parties
+
+  def all_parties
+    Party.where(user_id: id).or(Party.where(id: PartyUser.where(user_id: id).pluck(:party_id)))
+  end
```
Comment (LHJE) [inline app/models/user.rb:17]: ahh our old friend .or

**Review by LHJE — APPROVED** [review]
Looks good!  Once we get those merge conflicts resolved, this should make the whole thing much smoother

**Pull request merged**